File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/core/PathSelectContext.java
package cn.gjing.tools.swagger.core;

import cn.gjing.tools.swagger.SwaggerBean;
import com.google.common.base.Predicate;

/**
 * @author Gjing
 **/
@SuppressWarnings("all")
final class PathSelectContext {
    public Predicate<String> getPredicate(SwaggerBean swaggerBean, String pathPattern) {
        return PathSelectFactory.SELECT.create(swaggerBean.getPathType()).getPredicate(swaggerBean,pathPattern);
    }
}

File: tools-starter-swagger/src/main/java/cn/gjing/tools/swagger/core/PathSelectContext.scala
package cn.gjing.tools.swagger.core

import cn.gjing.tools.swagger.SwaggerBean
import com.google.common.base.Predicate

import scala.beans.BeanProperty

/**
 * @author Gjing
 **/
class PathSelectContext {
  def getPredicate(swaggerBean: SwaggerBean, pathPattern: String): Predicate[String] = {
    PathSelectFactory.SELECT.create(swaggerBean.getPathType).getPredicate(swaggerBean, pathPattern)
  }
}
